src/utils/utils.ts

```diff
@@ -26,8 +26,8 @@ export function escapeRegExp(str: string) {
  * @return the input text with comments removed.
  * Note the number lines of the output matches the input
  */
-export function stripComments(text: string, commentSign: string): string {
-    const pattern = '([^\\\\]|^)' + commentSign + '.*$'
+export function stripComments(text: string, commentSign: string = '%'): string {
+    const pattern = '(^|[^\\\\]|(?:(?<!\\\\)\\\\\\\\))' + commentSign + '.*$'
```

Comment:
I don't like the API. It makes things too complicated unnecessarily. Please delete the `commentSign` argument and use a literal regular expression. When we need to remove comments which begin other than `%`, let's define another function.